controllers/profiles/object_creators.go

```diff
@@ -0,0 +1,204 @@
+							Name:          "http",
+						}},
+					}},
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+// naiveApplyImageDeploymentMutateVisitor creates a visitor that mutates a vanilla Kubernetes Deployment to apply the given image in the first container
+func naiveApplyImageDeploymentMutateVisitor(image string) mutateVisitor {
+	return func(object client.Object) controllerutil.MutateFn {
+		return func() error {
+			object.(*appsv1.Deployment).Spec.Template.Spec.Containers[0].Image = image
+			return nil
+		}
+	}
+}
+
+func defaultDeploymentMutateVisitor(workflow *operatorapi.KogitoServerlessWorkflow) mutateVisitor {
+	return func(object client.Object) controllerutil.MutateFn {
+		return func() error {
+			if kubeutil.IsObjectNew(object) {
+				return nil
+			}
+			original, err := defaultDeploymentCreator(workflow)
+			if err != nil {
+				return err
+			}
+			object.(*appsv1.Deployment).Spec.Template.Spec.Volumes = make([]corev1.Volume, 0)
+			object.(*appsv1.Deployment).Spec.Template.Spec.Volumes = original.(*appsv1.Deployment).Spec.Template.Spec.Volumes
+			object.(*appsv1.Deployment).Spec.Template.Spec.Containers = make([]corev1.Container, 0)
+			object.(*appsv1.Deployment).Spec.Template.Spec.Containers = original.(*appsv1.Deployment).Spec.Template.Spec.Containers
+			object.(*appsv1.Deployment).Spec.Replicas = original.(*appsv1.Deployment).Spec.Replicas
+			object.(*appsv1.Deployment).Labels = original.GetLabels()
+			return nil
+		}
+	}
+}
+
+// defaultServiceCreator is an objectCreator for a basic Service aiming a vanilla Kubernetes Deployment.
+// It maps the default HTTP port (80) to the target Java application webserver on port 8080.
+func defaultServiceCreator(workflow *operatorapi.KogitoServerlessWorkflow) (client.Object, error) {
+	lbl := labels(workflow)
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workflow.Name,
+			Namespace: workflow.Namespace,
+			Labels:    lbl,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: lbl,
+			Ports: []corev1.ServicePort{{
+				Protocol:   corev1.ProtocolTCP,
+				Port:       defaultHTTPServicePort,
+				TargetPort: intstr.FromInt(defaultHTTPWorkflowPort),
+			}},
+		},
+	}
+	return service, nil
+}
+
+func defaultServiceMutateVisitor(workflow *operatorapi.KogitoServerlessWorkflow) mutateVisitor {
+	return func(object client.Object) controllerutil.MutateFn {
+		return func() error {
+			if kubeutil.IsObjectNew(object) {
+				return nil
+			}
+			original, err := defaultServiceCreator(workflow)
+			if err != nil {
+				return err
+			}
+			object.(*corev1.Service).Spec.Ports = original.(*corev1.Service).Spec.Ports
+			object.(*corev1.Service).Labels = original.GetLabels()
+			return nil
+		}
+	}
+}
+
+// workflowSpecConfigMapCreator creates a new ConfigMap that holds the definition of a workflow specification.
+func workflowSpecConfigMapCreator(workflow *operatorapi.KogitoServerlessWorkflow) (client.Object, error) {
```

Comment:
wdyt to add a extra check here and use the sdk-go to parse the workflow?

Or this step will be handled by the validation that @amadhusu is working on?

Reply:
We are not doing validations here, the spec is coming from the CR, so we expect that the admission controller does that job. Yes, the @amadhusu 's work.